packages/wrangler/src/sourcemap.ts

```diff
@@ -1,10 +1,60 @@
 import assert from "node:assert";
+import fs from "node:fs";
+import url from "node:url";
+import type { Options } from "@cspotcode/source-map-support";
 import type Protocol from "devtools-protocol";
 
+function maybeGetFile(filePath: string | URL) {
+	try {
+		return fs.readFileSync(filePath, "utf8");
+	} catch (e: unknown) {
+		const notFound =
+			typeof e === "object" && e !== null && "code" in e && e.code === "ENOENT";
+		if (!notFound) throw e;
+	}
+}
+
+export type RetrieveSourceMap = NonNullable<Options["retrieveSourceMap"]>;
```

Comment:
NIT: I found this type a bit confusing to begin with. I would propose we call it `RetrieveSourceMapFunction` or `RetrieveSourceMapMethod`.